Dockerfile

```diff
@@ -24,30 +20,29 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     && apt-get clean \
     && rm -rf /var/lib/apt/lists/*
 
-# Install pnpm
-RUN npm install -g pnpm@9
 
-# Install Go from the official golang image
-COPY --from=golang:alpine /usr/local/go/ /usr/local/go/
-ENV PATH="/usr/local/go/bin:${PATH}"
+ENV GO_VERSION=1.21.1
+RUN curl -sL https://go.dev/dl/go$GO_VERSION.linux-amd64.tar.gz -o go$GO_VERSION.linux-amd64.tar.gz && \
+  tar -C /usr/local/ -xzvf go$GO_VERSION.linux-amd64.tar.gz
 
-# Install web3 cli
-RUN curl -LSs https://raw.githubusercontent.com/gochain/web3/master/install.sh | sh
```

Comment:
gud, this installation has occasionally failed for my setup

Reply:
oh really?